gcc/rust/privacy/rust-reachability.cc

```diff
@@ -53,7 +53,13 @@ ReachabilityVisitor::visit_generic_predicates (
 	  TyTy::BaseType *generic_ty = nullptr;
 	  auto ok = ty_ctx.lookup_type (generic->get_mappings ().get_hirid (),
 					&generic_ty);
-	  rust_assert (ok);
+
+	  // FIXME: This fails lookup on the `Self` generic parameter of the
+	  // trait definition. We can skip it for now, but that's not a real fix
+	  if (!ok)
```

Comment:
This is interesting you should put a debug line in to see what the ID is here and compare with what it is over in rust-hir-trait-resolve.h line 138 and ou should be able to call self->debug() we might be missing an insert call